testing/src/main/java/io/grpc/testing/TestClientCall.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2014, Google Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *    * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *    * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ *    * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.grpc.testing;
+
+import io.grpc.ClientCall;
+import io.grpc.Metadata;
+
+/**
+ * TestClientCall is a class that is designed for use in tests.  It is designed to be used in
+ * places where a scriptable call is necessary.  By default, all methods are noops, and designed to
+ * be overriden.
+ */
+public class TestClientCall<ReqT, RespT> extends ClientCall<ReqT, RespT> {
```

Comment:
The problem with base is that this is not a normal valid implementation. And implementations normally extend the base class, but implementations normally wouldn't extend this class.

I think you explained quite reasonably why `ClientInterceptors` is not using this class, independent of its name. Since this _is_ a noop implementation, I think we have to be okay with multiple such implementations, at least for now.

I mentioned that we'll maybe want a ClientCall version that throws UnsupportedOperationException for all methods, and at some other time we may make a Fake/Mock version. All of these various versions are "Test" implementations.